service/src/main/java/org/broadinstitute/listener/config/SamInspectorProperties.java

```diff
@@ -1,3 +1,42 @@
 package org.broadinstitute.listener.config;
 
-public record SamInspectorProperties(String samUrl, String samResourceId, String samResourceType) {}
+public final class SamInspectorProperties {
```

Comment:
any reason why converting this into a class instead of having it as a `record`? if it's just for default value, I think we can set it in application.yml?

Reply:
As a record it'd require the property to be set in the application.yml explicitly.  I wanted the value to be truly optional, that is, not required at all in the application.yml so existing config works with this version. However, if this is not a concern, and we are certain all services that use this version will update their config, I can revert it to a record, make the property required, and add it to the application.yml file.

Reply:
I think all the listener configuration comes in the form of environment variables, e.g. setting `LISTENER_SAMINSPECTORPROPERTIES_SAMRESOURCEID=...` at listener deployment time. In other words, there aren't any clients specifying their own `application.yml`. So it seems safe to put `samAction: write` in application.yml (we can test).

Reply:
yea, I like having all configs in application.yml just so I dont have to also look at other places about optional configs